app/controllers/volunteers_controller.rb

```diff
@@ -255,8 +255,8 @@ def home
     @open_shift_count = ScheduleChain.open_in_regions(current_volunteer.region_ids).length
 
     #Upcoming pickup list
-    @upcoming_pickups = Log.group_by_schedule(Log.upcoming_for(current_volunteer.id))
-    @sncs_pickups = Log.group_by_schedule(Log.needing_coverage(current_volunteer.region_ids,7,10))
+    @upcoming_pickups = Shift.build_shifts(Log.upcoming_for(current_volunteer.id))
+    @sncs_pickups = Shift.build_shifts(Log.needing_coverage(current_volunteer.region_ids,7,10))
```

Comment:
Args to `Log::needing_coverage`. Looks like `7` is `days_away` and `10` is a `limit` to apply.

```ruby
  def self.needing_coverage(region_id_list=nil, days_away=nil, limit=nil)
    unless region_id_list.nil?
      if days_away.nil?
        Log.where("\"when\" >= ?",Time.zone.today).where(:region_id=>region_id_list).order("logs.when").limit(limit).reject{ |l| l.covered? }
      else
        Log.where("\"when\" >= ? AND \"when\" <= ?",Time.zone.today,Time.zone.today+days_away).where(:region_id=>region_id_list).order("logs.when").limit(limit).reject{ |l| l.covered? }
      end
    else
      if days_away.nil?
        Log.where("\"when\" >= ?",Time.zone.today).order("logs.when").limit(limit).reject{ |l| l.covered? }
      else
        Log.where("\"when\" >= ? AND \"when\" <= ?",Time.zone.today,Time.zone.today+days_away).order("logs.when").limit(limit).reject{ |l| l.covered? }
      end
    end
  end
```